packages/common/src/index.ts

```diff
@@ -359,6 +360,58 @@ export default class Common {
     }
   }
 
+  /**
+   * Internal helper function to calculate a fork hash
+   * @param hardfork Hardfork name
+   * @returns Fork hash as hex string
+   */
+  _calcForkHash(hardfork: string) {
+    const genesis = Buffer.from(this.genesis().hash.substr(2), 'hex')
+
+    function pad(str: string, max: number): string {
+      return str.length < max ? pad('0' + str, max) : str
+    }
+
+    let hfBuffer = Buffer.alloc(0)
+    let prevBlock = 0
+    for (const hf of this.hardforks()) {
+      const block = hf.block
+
+      // Skip for chainstart (0), not applied HFs (null) and
+      // when already applied on same block number HFs
+      if (block !== 0 && block !== null && block !== prevBlock) {
+        const hfBlockBuffer = Buffer.from(pad(block.toString(16), 16), 'hex')
+        hfBuffer = Buffer.concat([hfBuffer, hfBlockBuffer])
+      }
+
+      if (hf.name === hardfork) break
+      prevBlock = block
+    }
+    const inputBuffer = Buffer.concat([genesis, hfBuffer])
+
+    // CRC32 delivers result as signed (negative) 32-bit integer,
+    // convert to hex string
+    const forkhash = new Number(crc32Buffer(inputBuffer) >>> 0).toString(16)
```

Comment:
What does this CRC lib do here? (sorry, I'm short on time to research it myself before my next break)

Reply:
CRC-32 is the chosen checksum algorithm from the `forkid` [EIP-2124](https://eips.ethereum.org/EIPS/eip-2124). The `>>>` bitshift operation converts the negative number result from the library into a positive 32-bit signed integer, this is then further converted to a buffer.

Reply:
All fork hash calculation comparisons with the static values are correct, so this should be hopefully some convincing evidence that this is correct.

Reply:
Right, thanks for the context!